src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Refactor to more readable approach and make more clear variable names, also use `map` to add `mother` field for every child for further usage :
```js
 const children = people
    .filter((child) => people.some((mother) => mother.name === child.mother))
    .filter((child) => (onlyWithSon ? child.sex === 'm' : true))
    .map((child) => ({
      ...child,
      mother: people.find((mother) => mother.name === child.mother),
    }));
```